extension.html

```diff
@@ -55,6 +55,39 @@ <h2>Introduction</h2>
       </p>
     </section>
+    <section>
+      <h2>Extensions to the Pointer Event types</h2>
+      <p>Below is a new pointer event type as well as a modification to an existing event in the <a href="https://w3c.github.io/pointerevents/index.html#pointer-event-types">existing pointer event types</a>.</p>
+      <section>
+        <h3>The <dfn><code>pointermove</code> event</dfn></h3>
+        <p>A user agent MUST <a href="#firing-events-using-the-pointerevent-interface">fire a pointer event</a> named <code>pointermove</code> when a pointer changes button state.
+        Additionally one <code>pointermove</code> MUST be fired when pointer changes coordinates, pressure, tangential pressure, tilt, twist, or
+        contact geometry (e.g. <code>width</code> and <code>height</code>) and the circumstances produce no other pointer events defined in this specification.
+        This <code>pointermove</code> that is not caused by button state change MUST be fired only once right before firing an event caused by button state change or
+        <a href="https://html.spec.whatwg.org/#event-loop-processing-model">animation frame callbacks</a>. If there are multiple changes to these attributes
+        the <code>pointermove</code> events before reaching animation frame callback or a button state change event,
+        these events will get coalesced and will be exposed via <a data-lt="getCoalescedEvents">getCoalescedEvents</a> API for the single dispatched <code>pointermove</code> event.
+        The final coordinates of such events should be used for finding the target of the event.
+      </section>
+      <section>
+        <h3>The <dfn><code>pointerrawmove</code> event</dfn></h3>
+        <p>A user agent MUST <a href="https://w3c.github.io/pointerevents/index.html#firing-events-using-the-pointerevent-interface">fire a pointer event</a>
+           named <code>pointerrawmove</code> when a pointing device attribute
+           (i.e. button state, coordinates, pressure, tangential pressure, tilt, twist, or contact geometry) is changed.
+           As oppose to <code>pointermove</code> which is delayed until an animation frame reached or button change events happen,
+           <code>pointerrawmove</code> will get fired as soon as an attribute is changed
+           and as frequent as the javascript can handle the events.
+           Because of that the <code>target</code> of <code>pointerrawmove</code> events might be different from the <code>pointermove</code> events
+           due to the fact that <code>pointermove</code> events are being coalesced and the final position of the event
+           which is used for finding the <code>target</code> could be different from its coalesced events.
+           Note that if the javascript handlers are slow, <code>pointerrawmove</code> events can also get coalesced.
```

Comment:
"Note that if the javascript handlers are slow, pointerrawmove events can also get coalesced."  what does this actually mean. What is slow JS? handler taking more than 16ms? handler forcing rAF to happen later? or what? And what is the reason for this coalescing? If the page wants raw events, it should get raw events. That would be simpler and easier to understand API.

Reply:
Let's discussion a little more about this. Just to make sure whether we should change the design and logic around this event or just the text here.

I agree that when the page does express the desire to get the pointerrawmove events then presumably they have a fast handler that doesn't take long. But imagine a case that the app is not well written and takes say a few milliseconds to handle even a pointerrawmove and I have a 1000Hz mouse and I'm interacting with the page. So in this case user agent is sending 1000 move events to the page (assuming mouse is moving around) or 1 event every millisecond. So user agent either should drop these raw move events or queue them forever but the queue is only growing in that case (as the incoming events are faster than the handled rate). So user agent should either drop some of the rawmove events or coalesce them and expose those coalesced events with getCoalescedEvents APIs. right?

Reply:
But if events are coalesced, when does that happen? When is it ok for UA to coalesce?
Is it ok to fire just one pointerrawmove per frame and coalesce the rest. Should the spec be a bit more precise. Something like... if UA detects that rendering will be updated and there are pending pointerrawmoves, fire just one event and coalesce the rest in the queue?

Reply:
I see. So in general you are okay with coalescing the rawmove events. The only problem is a reasonable definition around it and when to do the coalescing. Let me take another stab at the text.

Reply:
Sorry for the delay. I pushed a new commit. I used this sentence here instead: "Note that if UA detects the thread running javascript handlers for this event is not idle, pointerrawmove events will get coalesced and they will all be delivered as coalesced events of one pointerrawmove event as soon as the thread goes idle". Is that more precise?

Reply:
Rather vague still, since "thread" isn't defined. If we want to define this properly, it should probably talk about tasks and task queues.
Also, if there are pending idleCallbacks, will those cause coalescing?

Reply:
@smaug---- How about now?